zziplib: -lzzip doesn't work on darwin

**Describe the bug**

texlive core big bin builds failed on macOS.

```
ld: library not found for -lzzip
```

It seems failed after zziplib updated to 0.13.72.

```
...
ranlib libwindow.a
make[3]: Leaving directory '/private/tmp/nix-build-texlive-core-big.bin-2020.drv-0/texlive-20200406-source/WorkDir/texk/web2c/window'
clang -DHAVE_CONFIG_H -I. -I../../../texk/web2c -I./w2c  -I/nix/store/1dj801rf66ypq2qx8ssvlfdf4n4w5v5a-texlive-bin-2020/include  -DMFNOWIN  -Wimplicit -Wreturn-type -Wdeclaration-after-statement -Wno-unknown-pragmas -g -O2 -MT mf_nowin-mfextra.o -MD -MP -MF .deps/mf_nowin-mfextra.Tpo -c -o mf_nowin-mfextra.o `test -f 'mfextra.c' || echo '../../../texk/web2c/'`mfextra.c
mv -f .deps/mf_nowin-mfextra.Tpo .deps/mf_nowin-mfextra.Po
/nix/store/kl013dwzxf67l4pfrx6x2d5q6pq9gf36-bash-4.4-p23/bin/bash ../../../texk/web2c/web2c/convert mflua
/nix/store/kl013dwzxf67l4pfrx6x2d5q6pq9gf36-bash-4.4-p23/bin/bash ../../../texk/web2c/web2c/convert mfluajit
/nix/store/kl013dwzxf67l4pfrx6x2d5q6pq9gf36-bash-4.4-p23/bin/bash ./libtool  --tag=CC   --mode=link clang -Wimplicit -Wreturn-type -Wdeclaration-after-statement -Wno-unknown-pragmas -g -O2 -export-dynamic  -o luatex luatexdir/luatex-luatex.o mplibdir/luatex-lmplib.o libluatex.a libluatexspecific.a libluatex.a libff.a libluamisc.a libluasocket.a libluaffi.a libluapplib.a /private/tmp/nix-build-texlive-core-big.bin-2020.drv-0/texlive-20200406-source/WorkDir/libs/lua53/libtexlua53.la libmplibcore.a -L/nix/store/3111czkmv34yw15mf83iw36hlq2gv96j-zziplib-0.13.72/lib -L/nix/store/2ca4w3gg0hdbws0jdhxm2rbbbs3v8n5n-zlib-1.2.11/lib -lzzip -lz -L/nix/store/2ca4w3gg0hdbws0jdhxm2rbbbs3v8n5n-zlib-1.2.11/lib -L/nix/store/fsdrfs0njd12j3spq2cqhn8kzwp8vvj1-libpng-apng-1.6.37/lib -lpng16 -lz -lz lib/lib.a -L/nix/store/1dj801rf66ypq2qx8ssvlfdf4n4w5v5a-texlive-bin-2020/lib -lkpathsea libmputil.a libunilib.a libmd5.a
libtool: link: clang -Wimplicit -Wreturn-type -Wdeclaration-after-statement -Wno-unknown-pragmas -g -O2 -o luatex luatexdir/luatex-luatex.o mplibdir/luatex-lmplib.o  libluatexspecific.a libluatex.a libff.a libluamisc.a libluasocket.a libluaffi.a libluapplib.a /private/tmp/nix-build-texlive-core-big.bin-2020.drv-0/texlive-20200406-source/WorkDir/libs/lua53/.libs/libtexlua53.a libmplibcore.a -L/nix/store/3111czkmv34yw15mf83iw36hlq2gv96j-zziplib-0.13.72/lib -L/nix/store/2ca4w3gg0hdbws0jdhxm2rbbbs3v8n5n-zlib-1.2.11/lib -lzzip -L/nix/store/fsdrfs0njd12j3spq2cqhn8kzwp8vvj1-libpng-apng-1.6.37/lib /nix/store/fsdrfs0njd12j3spq2cqhn8kzwp8vvj1-libpng-apng-1.6.37/lib/libpng16.dylib -lz lib/lib.a -L/nix/store/1dj801rf66ypq2qx8ssvlfdf4n4w5v5a-texlive-bin-2020/lib /nix/store/1dj801rf66ypq2qx8ssvlfdf4n4w5v5a-texlive-bin-2020/lib/libkpathsea.dylib libmputil.a libunilib.a libmd5.a
ld: library not found for -lzzip
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [Makefile:7200: luatex] Error 1
make[2]: Leaving directory '/private/tmp/nix-build-texlive-core-big.bin-2020.drv-0/texlive-20200406-source/WorkDir/texk/web2c'
make[1]: *** [Makefile:16966: all-recursive] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-texlive-core-big.bin-2020.drv-0/texlive-20200406-source/WorkDir/texk/web2c'
make: *** [Makefile:5089: all] Error 2
error: builder for '/nix/store/7nm748rcxm4ilv1cpr0x6mr6wpa5w1m7-texlive-core-big.bin-2020.drv' failed with exit code 2
error: 1 dependencies of derivation '/nix/store/rac78ydcabgfcgazkx21mac3mag536xf-texlive-combined-full-2020.20210109.drv' failed to build
error: 1 dependencies of derivation '/nix/store/s52gkxwgz2fa83zfirqwgkscbgf5jw3k-home-manager-fonts.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5sh24rbazavw6r326qjsc5az8v1h0qsc-home-manager-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/2jw1p5wqr6g8vzk6mndngsqw6wgnf2qs-home-manager-generation.drv' failed to build
```

Full log:

https://gist.github.com/nasyxx/a18951d0e3392cc88a986a669da7b398

**To Reproduce**
Steps to reproduce the behavior:
1. `nix-shell -p texlive.combined.scheme-full`


**Metadata**

 - system: `"x86_64-darwin"`
 - host os: `Darwin 20.3.0, macOS 10.16`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.4pre20210317_8a5203d`
 - channels(Nasy): `"home-manager, nasy-pynix, nixpkgs"`
 - nixpkgs: `/Users/Nasy/.nix-defexpr/channels/nixpkgs`

--- a/pkgs/development/libraries/zziplib/default.nix
+++ b/pkgs/development/libraries/zziplib/default.nix
@@ -1,6 +1,7 @@
 { lib
 , stdenv
 , fetchFromGitHub
+, fetchpatch
 , cmake
 , perl
 , pkg-config
@@ -20,6 +21,22 @@ stdenv.mkDerivation rec {
     rev = "v${version}";
     hash = "sha256-Ht3fBgdrTm4mCi5uhgQPNtpGzADoRVOpSuGPsIS6y0Q=";
   };
+
+  patches = [
+    # apply https://github.com/gdraheim/zziplib/pull/113
+    (fetchpatch {
+      url = "https://github.com/gdraheim/zziplib/commit/82a7773cd17828a3b0a4f5f552ae80c1cc8777c7.diff";
+      sha256 = "0ifqdzxwb5d19mziy9j6lhl8wj95jpxzm0d2c6y3bgwa931avd3y";
+    })
+    (fetchpatch {
+      url = "https://github.com/gdraheim/zziplib/commit/1cd611514c5f9559eb9dfc191d678dfc991f66db.diff";
+      sha256 = "11w9qa46xq49l113k266dnv8izzdk1fq4y54yy5w8zps8zd3xfny";
+    })
+    (fetchpatch {
+      url = "https://github.com/gdraheim/zziplib/commit/e47b1e1da952a92f917db6fb19485b8a0b1a42f3.diff";
+      sha256 = "0d032hkmi3s3db12z2zbppl2swa3gdpbj0c6w13ylv2g2ixglrwg";
+    })
+  ];
 
   nativeBuildInputs = [
     cmake

